hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -158,14 +153,15 @@ public Record getRecord(Data key) {
     }
 
     @Override
-    public Record putReplicatedRecord(Record replicatedRecord, long nowInMillis) {
+    public Record putReplicatedRecord(Record replicatedRecord, long nowInMillis,
+                                      boolean populateIndexes) {
         Record newRecord = createRecord(replicatedRecord, nowInMillis);
         markRecordStoreExpirable(replicatedRecord.getTtl(), replicatedRecord.getMaxIdle());
         // Note that newRecord may not know its key yet, key assignment
         // is done after storage put in some cases so better to
         // use key of replicatedRecord when doing storage.put
         storage.put(replicatedRecord.getKey(), newRecord);
-        mutationObserver.onReplicationPutRecord(newRecord.getKey(), newRecord);
+        mutationObserver.onReplicationPutRecord(newRecord.getKey(), newRecord, !populateIndexes);
```

Comment:
Here we change the meaning of `populateIndexes`. `onReplicationPutRecord` accepts it as a `backup` parameter and `IndexingMutationObserver` uses it internally as the original meaning of the `populateIndex`. `populateIndex` here can be false for other reasons too. 

Maybe passing around a `MutationEvent` parameter object carrying index and backup info (we need to think about others too) would be more flexible and extensible.

Reply:
changed flag's name from `backup` to `populateIndex` for `onReplicationPutRecord`.